### PR TITLE
Add application code

### PR DESCRIPTION
**Pull request by brinxmat**
First approach to delivering the ontologies as static files.

### CODE REVIEW

File: ontology-service/src/main/java/no/sikt/nva/OntologyController.java [hunk 1]
```diff
@@ -1,18 +1,60 @@
...
-    public String getFile(@PathVariable("file") String file) {
-        logger.info("Get called with value: " + file);
-        return "ok";
+    private String getResourceAsString(String file, String extension) throws IOException {
+        Optional<InputStream> inputStream = loader.getResourceAsStream(file + EXTENSION_SEPERATOR + extension);
+        if (inputStream.isPresent()) {
+            return new String(inputStream.get().readAllBytes());
+        }
+        logger.warn(String.format("Requested file \"%s.%s\" was not found", file, extension));
+        return null;
```
Comment (janbj) [inline ontology-service/src/main/java/no/sikt/nva/OntologyController.java:58]: Why are we returning null?

File: ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java [hunk 2]
```diff
@@ -5,25 +5,36 @@
...
+import java.net.URI;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.zalando.problem.Status.NOT_ACCEPTABLE;
 
 public class OntologyControllerTest {
 
     private static MicronautLambdaHandler handler;
     private static final Context lambdaContext = new MockLambdaContext();
+    public static ObjectMapper objectMapper;
```
Comment (janbj) [inline ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java]: public?

File: ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java [hunk 3]
```diff
@@ -36,11 +47,72 @@ public static void cleanupSpec() {
...
+
+    @Test
+    void shouldReturnNTriples() {
+        AwsProxyRequest request = new AwsProxyRequestBuilder("/test", HttpMethod.GET.toString())
+                .header(HttpHeaders.ACCEPT, "application/ntriples")
                 .build();
         AwsProxyResponse response = handler.handleRequest(request, lambdaContext);
-        Assertions.assertEquals(HttpStatus.OK.getCode(), response.getStatusCode());
+        assertEquals(HttpStatus.OK.getCode(), response.getStatusCode());
+        assertEquals("_:b0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> \"test\" .",
```
Comment (janbj) [inline ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java]: NTRIPLES_RESPONSE_BODY constant?

File: ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java [hunk 4]
```diff
@@ -36,11 +47,72 @@ public static void cleanupSpec() {
...
+    void shouldReturnNotAcceptable() throws JsonProcessingException {
+        AwsProxyRequest request = new AwsProxyRequestBuilder("/test", HttpMethod.GET.toString())
+                .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
+                .build();
+        AwsProxyResponse response = handler.handleRequest(request, lambdaContext);
+        assertEquals(HttpStatus.NOT_ACCEPTABLE.getCode(), response.getStatusCode());
+        assertEquals("application/problem+json", getFirstMultiValueHeader(response));
+        var body = objectMapper.readValue(response.getBody(), Problem.class);
+        assertEquals(NOT_ACCEPTABLE, body.getStatus());
+        assertEquals(URI.create("about:blank"), body.getType());
```
Comment (janbj) [inline ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java]: Use constant describing why this value is used

File: ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java [hunk 5]
```diff
@@ -36,11 +47,72 @@ public static void cleanupSpec() {
     }
 
     @Test
-    void shouldReturnOk() {
-        AwsProxyRequest request = new AwsProxyRequestBuilder("/file", HttpMethod.GET.toString())
-                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+    void shouldReturnTurtle() {
```
Comment (axthosarouris) [inline ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java]: shouldReturnTurtleWhenAcceptHeaderIsApplicationTurtle. In general add the "when" condition in all test names

File: ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java [hunk 6]
```diff
@@ -36,11 +47,72 @@ public static void cleanupSpec() {
...
+        AwsProxyRequest request = new AwsProxyRequestBuilder("/test", HttpMethod.GET.toString())
+                .header(HttpHeaders.ACCEPT, "application/turtle")
+                .build();
+        AwsProxyResponse response = handler.handleRequest(request, lambdaContext);
+        assertEquals(HttpStatus.OK.getCode(), response.getStatusCode());
+        assertEquals("_:b0 a \"test\" .", response.getBody());
+    }
+
+    @Test
+    void shouldReturnNTriples() {
```
Comment (axthosarouris) [inline ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java]: ditto

File: ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java [hunk 7]
```diff
@@ -36,11 +47,72 @@ public static void cleanupSpec() {
...
                 .build();
         AwsProxyResponse response = handler.handleRequest(request, lambdaContext);
-        Assertions.assertEquals(HttpStatus.OK.getCode(), response.getStatusCode());
+        assertEquals(HttpStatus.OK.getCode(), response.getStatusCode());
+        assertEquals("_:b0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> \"test\" .",
+                response.getBody());
+    }
+
+    @Test
+    void shouldReturnNotFound() {
```
Comment (axthosarouris) [inline ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java]: add the condition in the tests.

File: ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java [hunk 8]
```diff
@@ -36,11 +47,72 @@ public static void cleanupSpec() {
...
+    }
+
+    @Test
+    void shouldReturnJsonLd() {
+        AwsProxyRequest request = new AwsProxyRequestBuilder("/test", HttpMethod.GET.toString())
+                .header(HttpHeaders.ACCEPT, "application/ld+json")
+                .build();
+        AwsProxyResponse response = handler.handleRequest(request, lambdaContext);
+        assertEquals(HttpStatus.OK.getCode(), response.getStatusCode());
+        assertEquals("{\"@context\": {\"@vocab\": \"https://example.org/\"}, \"name\": \"test\"}",
```
Comment (axthosarouris) [inline ontology-service/src/test/java/no/sikt/nva/OntologyControllerTest.java]: Is this the same content as the file? Why not read the file here? You can import nva-commons:core in testImplementation to read from resources or you re-use the method `getResourceAsString`

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Review by axthosarouris — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**